src/java/org/apache/cassandra/metrics/MessagingMetrics.java

```diff
@@ -160,6 +163,25 @@ public void recordTotalMessageProcessingTime(Verb verb, InetAddressAndPort from,
     public void recordDroppedMessage(Message<?> message, long timeElapsed, TimeUnit timeUnit)
     {
         recordDroppedMessage(message.verb(), timeElapsed, timeUnit, message.isCrossNode());
+        
+        if (message.verb() == Verb.MUTATION_REQ && message.payload instanceof org.apache.cassandra.db.Mutation)
+        {
+            org.apache.cassandra.db.Mutation mutation = (org.apache.cassandra.db.Mutation) message.payload;
+            for (org.apache.cassandra.db.partitions.PartitionUpdate update : mutation.getPartitionUpdates())
+            {
+                String tableKey = update.metadata().keyspace + '.' + update.metadata().name;
+                DroppedMessageMetrics tableMetrics = droppedMutationsByTable.computeIfAbsent(tableKey, 
+                    k -> {
+                        DefaultNameFactory tableFactory = new DefaultNameFactory("DroppedMutations", tableKey);
```

Comment:
Nit: This feels a bit weird - we're creating the factory that wraps some const and tableName but then pass again the const and the tableName. Somehow I'd rather avoid passing in the `tableFactory` and just past the remaining two.